js_modules/dagster-ui/packages/ui-core/src/asset-graph/useAssetGraphData.tsx

```diff
@@ -63,11 +70,17 @@ export function useAssetGraphData(opsQuery: string, options: AssetGraphFetchScop
     [repoFilteredNodes],
   );
 
-  const fullGraphQueryItems = useMemo(() => (nodes ? buildGraphQueryItems(nodes) : []), [nodes]);
+  const fullGraphQueryItems = useMemo(
+    () => (nodes && options.needsFullData ? buildGraphQueryItems(nodes) : []),
+    [nodes, options.needsFullData],
+  );
 
   const fullAssetGraphData = useMemo(
-    () => (fullGraphQueryItems ? buildGraphData(fullGraphQueryItems.map((n) => n.node)) : null),
-    [fullGraphQueryItems],
+    () =>
+      fullGraphQueryItems && options.needsFullData
+        ? buildGraphData(fullGraphQueryItems.map((n) => n.node))
+        : null,
+    [fullGraphQueryItems, options.needsFullData],
```

Comment:
Instead of making these optional, could we move them to a separate hook? It seems like they're not used further down so it'd be safe to throw these 10 lines up in the caller

Reply:
They rely on the same data from `ASSET_GRAPH_QUERY`, I can move it to a separate hook though because `useIndexedDBCachedQuery` will dedupe the requests but there will be some duplicated logic.